orangecontrib/prototypes/widgets/owsplit.py

```diff
@@ -0,0 +1,135 @@
+        if data.domain != self.new_data.domain:
+            data = data.transform(self.new_data.domain)
+        return self.new_data.transform(data.X)
+
+
+class OneHotStrings(SharedComputeValue):
+
+    def __init__(self, fn, feature):
+        super().__init__(fn)
+        self.feature = feature
+
+    def compute(self, _, transformed):
+        return transformed[:, self.feature]
+
+
+class OWSplit(OWWidget):
+    name = "Split"
+    description = "Split string variables to create discrete."
+    icon = "icons/Split.svg"
+    priority = 700
+
+    class Inputs:
+        data = Input("Data", Table)
+
+    class Outputs:
+        data = Output("Data", Table)
+
+    class Warning(OWWidget.Warning):
+        no_disc = Msg("Data contains only numeric variables.")
+
+    want_main_area = False
+    resizing_enabled = False
+
+    settingsHandler = DomainContextHandler()
+    attribute = ContextSetting(None)
+    splitter = ContextSetting(";")
+
+    def __init__(self):
+        super().__init__()
+        self.data = None
+
+        gui.lineEdit(
+            self.controlArea, self, "splitter",
+            orientation=Qt.Horizontal, box="Split by", callback=self.apply)
+
+        variable_select_box = gui.vBox(self.controlArea, "Variable")
+
+        gui.comboBox(variable_select_box, self, "attribute", label=None,
+                     orientation=Qt.Horizontal, searchable=True,
+                     callback=self.apply,
+                     model=DomainModel(valid_types=(StringVariable,
+                                                    DiscreteVariable)))
+
+    @Inputs.data
+    def set_data(self, data):
+        self.closeContext()
+        self.data = data
+
+        model = self.controls.attribute.model()
+        model.set_domain(data.domain if data is not None else None)
+        self.Warning.no_disc(shown=data is not None and not model)
+        if not model:
+            self.attribute = None
+            self.Outputs.data.send(None)
+            return
+        self.attribute = model[0]
+        self.openContext(data)
+        self.apply()
+
+    def apply(self):
+        var = self.data.domain[self.attribute]
+        values = self.data.get_column_view(self.attribute)[0]
+        if var.is_discrete:
+            values = values_to_string(values, var.values)
+
+        new_data, new_values = values_from_string(values, self.splitter)
+
+        new_columns = tuple(
+            Orange.data.ContinuousVariable(
+                v,
+                compute_value=OneHotStrings(values, values_from_string(values,
+                                                                       self.splitter))
+            ) for v in new_values)
+
+        new_domain = Orange.data.Domain(
+            self.data.domain.attributes + new_columns,
+            self.data.domain.class_vars, self.data.domain.metas
+        )
+        extended_data = self.data.transform(new_domain)
+        extended_data.X = np.hstack([self.data.X, new_data])
```

Comment:
This shouldn't be necessary. `self.data.transform(new_domain)` must compute values for the new attributes, otherwise something is wrong with `compute_value`.

It also won't work because tables need to be unlocked.